accio-cache/src/main/java/io/accio/cache/CacheManager.java

```diff
@@ -250,29 +265,38 @@ public void removeTempFile(PathInfo pathInfo)
         }
     }
 
-    public TaskInfo createTaskUtilDone(AccioMDL mdl)
+    public List<TaskInfo> createTaskUtilDone(AccioMDL mdl)
+    {
+        return createTask(mdl)
+                .thenApply(taskInfos -> taskInfos.stream()
+                        .map(taskInfo -> {
+                            tasks.get(taskInfo.getCatalogSchemaTableName()).waitUntilDone();
+                            return getTaskInfo(taskInfo.getCatalogSchemaTableName()).join();
+                        })
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(toList())).join();
+    }
+
+    public CompletableFuture<List<TaskInfo>> createTask(AccioMDL mdl)
     {
-        Optional<TaskInfo> taskInfoOptional = createTask(mdl)
-                .thenCompose(taskInfo -> {
-                    tasks.get(taskInfo.getTaskId()).waitUntilDone();
-                    return getTaskInfo(taskInfo.getTaskId());
-                })
-                .join();
-        return taskInfoOptional.orElseThrow(() -> new RuntimeException("Failed to create task"));
+        return supplyAsync(() ->
+                mdl.listCached().stream().map(cacheInfo -> createTask(mdl, cacheInfo).join()).collect(toList()));
     }
 
-    public CompletableFuture<TaskInfo> createTask(AccioMDL mdl)
+    public CompletableFuture<TaskInfo> createTask(AccioMDL mdl, CacheInfo cacheInfo)
     {
         return supplyAsync(() -> {
-            String taskId = randomUUID().toString();
-            TaskInfo taskInfo = new TaskInfo(taskId, mdl.getCatalog(), mdl.getSchema(), RUNNING, Instant.now());
-            Task task = new Task(taskInfo, refreshCache(mdl));
-            tasks.put(taskId, task);
+            CatalogSchemaTableName catalogSchemaTableName = new CatalogSchemaTableName(mdl.getCatalog(), mdl.getSchema(), cacheInfo.getName());
+            TaskInfo taskInfo = new TaskInfo(mdl.getCatalog(), mdl.getSchema(), cacheInfo.getName(), RUNNING, Instant.now());
+            // To fix flaky test, we pass value to tasks instead of a reference;
+            Task task = new Task(TaskInfo.copyFrom(taskInfo), handleCache(mdl, cacheInfo));
```

Comment:
Could you explain more ? Not sure which test will be flaky without using copyFrom

Reply:
During testing, we will verify that the status of TaskInfo is set to 'RUNNING.' However, it's important to note that the task might conclude before relaying the information to the client. Therefore, we need to capture a snapshot at the task creation stage to accurately assess its status.